### PR TITLE
implement ssim loss

### PR DESCRIPTION
**Pull request by edgarriba**

### CODE REVIEW

**Comment by m-zheng** [comment]
Hi,

Can I ask why you want to divide ssim loss value by 2 before being subtracted by 1? 
https://github.com/kornia/kornia/blob/0deb07968053a6f6dea38754b4167498ad59e7b7/kornia/losses/ssim.py#L48

Shouldn't it be just 1 - ssim rather than 1 - ssim / 2?

**Comment by edgarriba** [comment]
> Hi,
> 
> Can I ask why you want to divide ssim loss value by 2 before being subtracted by 1?
> 
> https://github.com/kornia/kornia/blob/0deb07968053a6f6dea38754b4167498ad59e7b7/kornia/losses/ssim.py#L48
> 
> Shouldn't it be just 1 - ssim rather than 1 - ssim / 2?

the intention was to regularize the gradients given that the ssim measure comes from two independent image samples. Open to discussion if there's a better approach.

**Comment by m-zheng** [comment]
Thanks for your reply.

If it is indented to regularise the gradients, why is it not other values?

Also, it is shown as  
`loss = torch.clamp(1. - ssim_map, min=0, max=1) / 2.`
 in (https://kornia.readthedocs.io/en/v0.1.2/_modules/torchgeometry/losses/ssim.html), which is different from what is implemented here.

**Comment by edgarriba** [comment]
this is a very old version of the code -- as far as I remember we modified that one after some users complained about the previous implementation and was agreed to update to what we have right now
